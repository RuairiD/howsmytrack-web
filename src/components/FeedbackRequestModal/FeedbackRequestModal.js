import React from 'react';

import { Modal } from 'antd';
import CreateFeedbackRequestForm from '../CreateFeedbackRequestForm/CreateFeedbackRequestForm';

type Props = {
    onCancel: () => void,
    isVisible: boolean,
};

class FeedbackRequestModal extends React.Component<Props> {
    /*
     * Component for displaying modal for submitting a feedback request.
     */
    render() {
        return (
            <Modal
                title="New Feedback Request"
                onCancel={this.props.onCancel}
                visible={this.props.isVisible}
                footer={null}
                destroyOnClose
            >
                <CreateFeedbackRequestForm />
            </Modal>
        );
    }
}

export default FeedbackRequestModal;
