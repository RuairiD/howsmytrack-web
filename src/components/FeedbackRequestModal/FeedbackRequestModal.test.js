import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackRequestModal from "./FeedbackRequestModal";

describe("FeedbackRequestModal", () => {
    it("renders a Modal containing a CreateFeedbackRequestForm", () => {
        const onCancel = () => {};
        const wrapper = shallow(
            <FeedbackRequestModal
                onCancel={onCancel}
                isVisible
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
