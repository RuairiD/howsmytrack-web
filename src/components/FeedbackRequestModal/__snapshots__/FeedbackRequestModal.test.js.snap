// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackRequestModal renders a Modal containing a CreateFeedbackRequestForm 1`] = `
<Modal
  className="responsive-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer={null}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[Function]}
  title="New Feedback Request"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form(UnwrappedCreateFeedbackRequestForm) />
</Modal>
`;
